handle/Timer.lua

```diff
@@ -1,30 +1,190 @@
 require('CeresStdLib.base.Native')
 require('CeresStdLib.base.Log')
-local free_timers = {}
+
+local Timer 		= {}
+local free_timers	= {}
+local MAX_PUSHES	= 10000		--	Maximum amount of times push is called before garbage collection.
+
+free_timers[0]		= 0
+free_timers[-1]		= 0
+
+free_timers.has = function()
+	if free_timers[0] <= 0 then return false end return true
+end
+
+free_timers.push = function(t)
+	if free_timers[-1] >= MAX_PUSHES then
+		collectgarbage()
+	end
+	if not free_timers[t] then
+		free_timers[0] 					= free_timers[0] + 1
+		free_timers[t] 					= true
+		free_timers[free_timers[0]]		= t
+		free_timers[-1]					= free_timers[-1] + 1
+		return true
+	end
+	return false
+end
+
+free_timers.pop = function()
+	local obj
+	if free_timers[0] > 0 then
+		obj = free_timers[free_timers[0]]
+		free_timers[obj] 	= nil
+		free_timers[0] 		= free_timers[0] - 1
+		return obj
+	end
+	return nil
+end
 
 replaceNative('CreateTimer', function()
-	k, v = next(free_timers)
-	if k ~= nil then
-		free_timers[k] = nil
-		return k
+	local obj
+	if free_timers.has() then
+		obj = free_timers.pop()
+	else
+		obj = Native.CreateTimer()
+	end
+	Timer[obj]      = {
```

Comment:
Are all of these fields necessary? Overall it feels a bit overcomplicated.
I still can merge it, and try myself at simplifying it, though.

Reply:
The fields describe the current state of the timer.
hasCallback indicates when a callback is added to the timer (via TimerStart). This is for the ResumeTimer crash fix.
inCallback indicates when the timer has expired and is now calling the callback.
callback is the callback function of the timer.